### PR TITLE
add python code generator for API classes

### PR DESCRIPTION
**Pull request by guillermooo**
Also, bump up analyzer package version. This is needed by the Python
code generator to output nicer-looking docstrings.

---

@bwilkerson I'm using this generator for a new version of a Sublime Text Dart plugin. It works, but it's incomplete, has many to-dos and probably some rough edges still. My plan is to test the generated classes as I need them. Otherwise it will take me too long.

Regardless, it would be great to know if there's any chance this can be merged into the sdk repo.

The generated Python code is included.

### CODE REVIEW

File: pkg/analysis_server/tool/spec/codegen_python.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+library codegen.python;
+
+import 'api.dart';
+
+/**
+ * Visitor specialized for generating Python code.
+ */
+class PythonCodegenVisitor extends HierarchicalApiVisitor {
+  /**
+   * Type references in the spec that are named something else in Dart.
```
Comment (bwilkerson) [inline pkg/analysis_server/tool/spec/codegen_python.dart]: "Dart" --> "Python"?

File: pkg/analysis_server/tool/spec/codegen_python.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  static const Map<String, String> _typeRenames = const {
+    'long': 'int',
+    'object': 'Map',
+    'String': 'str',
+  };
+
+  PythonCodegenVisitor(Api api) : super(api);
+
+  /**
+   * Convert the given [TypeDecl] to a Dart type.
```
Comment (bwilkerson) [inline pkg/analysis_server/tool/spec/codegen_python.dart]: "Dart" --> "Python"

File: pkg/analysis_server/tool/spec/codegen_python_protocol.dart [hunk 3]
```diff
@@ -0,0 +1,1119 @@
...
+ */
+class ToJsonSnippet extends ToJsonCode {
+  /**
+   * Callback that can be used to generate the code snippet, once the name
+   * of the [value] variable is known.
+   */
+  final ToJsonSnippetCallback callback;
+
+  /**
+   * Dart type of the [value] variable.
```
Comment (bwilkerson) [inline pkg/analysis_server/tool/spec/codegen_python_protocol.dart]: "Dart" --> "Python"?

**Comment by bwilkerson** [comment]
LGTM, though I don't know Python, so I can't comment on the code being generated.

I'm happy to have this be part of the SDK repo.

**Comment by mit-mit** [comment]
@guillermooo to proceed, can you sign the CLA (see above) and address the review comments, please? 

Thanks again for contributing!

**Comment by mit-mit** [comment]
@guillermooo can you look at the SLA, please?

**Comment by mit-mit** [comment]
Closing: we are not able to take this PR without a signed CLA. Free free to reopen if the CLA is signed!

**Pull request closed without merging**